tests/DeepCopyTest/Reflection/ReflectionHelperTest.php

```diff
@@ -29,6 +29,40 @@ public function testMaintainPropertiesKey()
 
         $this->assertSame($expectedProps, array_keys($actualProps));
     }
+
+    /**
+     * @dataProvider propertyDataProvider
+     *
+     * @param string $name
+     */
+    public function testGetProperty($name)
+    {
+        $object = new ReflectionHelperTestChild();
+        $property = ReflectionHelper::getProperty($object, $name);
+        $this->assertInstanceOf('\ReflectionProperty', $property);
+        $this->assertSame($name, $property->getName());
+    }
+
+    /**
+     * @return array
+     */
+    public function propertyDataProvider()
+    {
+        return [
+            'public property' => ['a4'],
+            'private property' => ['a9'],
+            'private property of ancestor' => ['a3']
```

Comment:
Unless I didn't understand what's going on properly, you're not supposed to access to the private properties of the parent entity